master/internal/task/allocation_test.go

```diff
@@ -100,6 +100,18 @@ func TestAllocation(t *testing.T) {
 						Devices: []device.Device{},
 					},
 				}
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStateAssigned,
+					false).Return(nil)
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStatePulling,
+					false).Return(nil)
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStateStarting,
+					false).Return(nil)
+				db.On("UpdateAllocationState", a.req.AllocationID,
+					model.AllocationStateRunning, false).Return(nil)
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStateTerminating,
+					false).Return(nil)
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStateTerminated,
+					false).Return(nil)
```

Comment:
**question:** would it be valuable to have some tests that tests for db failures?

Reply:
probably, this just mocks out the db and assumes everything goes ok. having a test that systematically fails each db call and checks the outcome would be great.